src/bundle/Controller/SearchController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUiBundle\Controller;
+
+use eZ\Publish\Core\Repository\SearchService;
+use EzSystems\EzPlatformAdminUi\Form\Data\Search\SearchData;
+use EzSystems\EzPlatformAdminUi\Form\Type\Search\SearchType;
+use EzSystems\EzPlatformAdminUi\Tab\Dashboard\PagerContentToDataMapper;
+use Pagerfanta\View\TwitterBootstrap4View;
+use Symfony\Component\HttpFoundation\Request;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use Pagerfanta\Pagerfanta;
+use eZ\Publish\Core\Pagination\Pagerfanta\ContentSearchAdapter;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+class SearchController extends Controller
+{
+    /** @var SearchService */
+    private $searchService;
+
+    /** @var PagerContentToDataMapper */
+    protected $pagerContentToDataMapper;
+
+    /** @var Router */
+    protected $router;
+
+    /**
+     * @param SearchService $searchService
+     * @param PagerContentToDataMapper $pagerContentToDataMapper
+     * @param RouterInterface $router
+     */
+    public function __construct(
+        SearchService $searchService,
+        PagerContentToDataMapper $pagerContentToDataMapper,
+        RouterInterface $router
+    ) {
+        $this->searchService = $searchService;
+        $this->pagerContentToDataMapper = $pagerContentToDataMapper;
+        $this->router = $router;
+    }
+
+    /**
+     * Renders the simple search form and search results.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function searchAction(Request $request): Response
+    {
+        $search = $request->query->get('search');
+        $limit = $search['limit'] ?? 10;
+        $page = $search['page'] ?? 1;
+        $query = $search['query'];
+
+        $form = $this->createForm(SearchType::class, new SearchData($limit, $page, $query), [
+            'method' => 'GET',
+            'csrf_protection' => false
+        ]);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var SearchData $data */
+            $data = $form->getData();
+
+            $query = new Query();
+            $query->filter = new Criterion\LogicalAnd(
+                [
+                    new Criterion\Visibility(Criterion\Visibility::VISIBLE),
+                    new Criterion\FullText($data->getQuery()),
+                ]
+            );
+            $query->sortClauses[] = new SortClause\DateModified(Query::SORT_ASC);
+
```

Comment:
Please actually use `$limit` for query (`$query->limit`). Otherwise it will return default number of records (AFAICS now it is 25).
Also there's a need for `$query->offset`.

I don't know if and how this works with Pagerfanta, but offset and limits are needed to avoid killing external SE handler (e.g. can be Solr on a separate server)